action/protocol/staking/handlers.go

```diff
@@ -476,8 +508,14 @@ func (p *Protocol) handleDepositToStake(ctx context.Context, act *action.Deposit
 			failureStatus: iotextypes.ReceiptStatus_ErrInvalidBucketType,
 		}
 	}
-
-	prevWeightedVotes := p.calculateVoteWeight(bucket, csm.ContainsSelfStakingBucket(act.BucketIndex()))
+	selfStake, err := isSelfStakeBucket(featureCtx, csm, act.BucketIndex())
```

Comment:
move to L532, don't change order of calls

Reply:
also required at line 518